### DISCUSSION TITLE
Switch onChange

### DISCUSSION THREAD

**Discussion by anton-johansson** [chakra-ui/chakra-ui#2192] [Q&A] [answered]
Hey peeps! I'm trying out the `Switch` component, but I'm having a hard time using `onChange`. How can I react to changes of the `isChecked` property? The `onChange` property fires a simple `FormEvent<any>` and I can't seem to reach the `isChecked` value... Ideas?

**Comment by with-heart** [comment, marked as answer]
Hi @anton-johansson! You'd want to use `e.target.checked`, as seen in this example: https://codesandbox.io/s/vibrant-johnson-lchqn?file=/src/App.tsx

**Reply by anton-johansson** [reply]
Aha, I think we're missing some Typescript support for this component. I had to get around it by forcing the `target` to `any`, like so:
```tsx

                            <Switch size="lg" isChecked={myValue} onChange={event => {
                                const target: any = event.target;
                                setMyValue(target.checked);
                            }}/>
```

This works for now, but might be something to consider fixing in the future! :)

**EDIT:** Thanks @with-heart!

**Reply by with-heart** [reply]
@anton-johansson That's interesting. Are you seeing that locally? I updated the CodeSandbox to inline the function and I'm able to access `event.target.checked` without casting.

**Reply by anton-johansson** [reply]
Yeah, I'm seeing this locally. However, I've wrapped `Switch` in my own component now (for other reasons), where I'm forcing the correct type on my own props:
```
    /** The event handler for changing the value. */
    onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
```

So it works well. But yeah, it could maybe avoid confusion if it's the correct type to start with. :)

**EDIT:** If I browse into `onChange` in your sandbox page, I see the correct prop. But when I do the same locally, in `vscode`, I end up at a React interface called `DOMAttributes`:

```
        // Form Events
        onChange?: FormEventHandler<T>;
```

I'm using Chakra `0.8.0`.

**Reply by with-heart** [reply]
Oh, that explains it. Yeah TypeScript support is spotty at best on v0. It matches your type on v1: 

https://github.com/chakra-ui/chakra-ui/blob/acc6e161518e77873023106d13126ddce45f4a8e/packages/checkbox/src/use-checkbox.ts#L60-L63

**Reply by anton-johansson** [reply]
Allright, cool! But v1 is not released, is it?

**Reply by ljosberinn** [reply]
No, but it's encouraged to be used when you're just starting to build something with Chakra or are in an early phase. We have migration docs here: https://next.chakra-ui.com/docs/migration - please ensure to remove your lockfile as well as node_modules when migrating. Also you might need to update TypeScript (3.8.5 is min req but realistically there is no reason to not be on latest), @types/react as well as remove emotion dependencies (we'll add this info to the docs too).

**Reply by anton-johansson** [reply]
Thanks people! I'll migrate to v1. Do we have any estimate on when v1 will be released? I feel kinda weird about using a "non-stable" (it's probably stable, but you catch my drift) package.

**EDIT:** I assume 0.8.0 isn't really "stable" either, so it probably doesn't really matter. :D But would be great with an estimate anyway!

**Reply by ljosberinn** [reply]
Yes, we're fairly far along v1 so there shouldn't be any major bugs left.
At the current time it looks like we're seeing another rc to make sure everything  we recently changed is stable and then release 👀
